webpack.config.js

```diff
@@ -0,0 +1,155 @@
+          loader: 'file-loader',
+          query: {
+            name: 'static/media/[name].[hash:8].[ext]'
+          }
+        },
+        {
+          test: /\.svg$/,
+          loader: 'svg-sprite-loader',
+          include: path.join(SOURCE_DIR, 'images'),
+          options: {
+            symbolId: filePath => `icon-${path.basename(filePath, '.svg')}`
+          }
+        },
+        {
+          test: /\.js$|jsx?$/,
+          loader: 'babel-loader',
+          options: {
+            cacheDirectory: true,
+            presets: [babelEnv({modules: false}), '@babel/preset-react'],
+            plugins: [
+              '@babel/plugin-transform-flow-strip-types',
+              '@babel/plugin-proposal-object-rest-spread',
+              '@babel/plugin-proposal-class-properties',
+              'react-hot-loader/babel'
+            ],
+            env: {
+              test: {
+                presets: [babelEnv({modules: 'auto'})]
+              }
+            }
+          }
+        }, {
+          test: /\.scss$/,
+          resolve: {
+            extensions: ['.scss', '.sass']
+          },
+          use: [
+            'style-loader',
+            'css-loader',
+            {
+              loader: 'postcss-loader',
+              options: {
+                ident: 'postcss',
+                plugins: [
+                  require('autoprefixer')({
+                    browsers: [
+                      'last 2 versions',
+                      'OperaMini >= 5',
+                      'Android >= 4',
+                      'Chrome >= 28',
+                      'Safari >= 7'
+                    ],
+                    cascade: false
+                  })]
+              }
+            }, {
+              loader: 'sass-loader',
+              options: {
+                includePaths: [
+                  SOURCE_DIR,
+                  path.resolve(__dirname, 'node_modules')
+                ]
+              }
+            }],
+          include: path.resolve(__dirname)
+        }
+      ]
+    },
+    externals: {
+      'html_beautify': {
+        root: 'html_beautify'
+      },
+      'hljs': 'hljs'
+    },
+    watch: true,
+    mode: 'development',
+    entry: {
+      [path.basename(jsEntry)]: jsEntry
+    },
+    output: {
+      filename: '[name].bundle.js',
+      path: jsOutput,
+      publicPath: '/'
+    },
+    resolve: {
+      extensions: ['.js', '.jsx'],
+      modules: [
+        SOURCE_COMPONENTS_DIR,
+        SOURCE_DOCS_DIR,
+        'node_modules'
+      ],
+      alias: {
+        'react-dom': '@hot-loader/react-dom'
+      }
+    },
+    plugins: [
+      new webpack.DefinePlugin({
+        'process.env.NODE_ENV': JSON.stringify(IS_PRODUCTION ? 'production' : 'development')
+      }), new HtmlWebpackPlugin({
+        template: path.join(SOURCE_DOCS_DIR, 'index.html')
+      })
+    ],
+    devtool: IS_PRODUCTION ? 'source-map' : 'eval',
+    optimization: {
```

Comment:
Do we need optimization for production? I'd rather keep in consistent with local env

Reply:
it's what we already have in the main config right now. lets leave it as is, because i wanna introduce proper build step some day and we will need it anyway

Reply:
Yeah for build step I agree it's necessary, however for serving standalone styleguide not that much

Reply:
But why custom config? Just use mode.